src/js/components/projects/createForm.tsx

```diff
@@ -86,160 +91,145 @@ const ProjectForm = ({
-            onClick={doGetBranches}
-          />
-          {i18n.t('to refresh this list or try')}{' '}
-        </>
-      )}
-      {invalidInput && (
-        <Button
-          label={i18n.t('starting from a new branch.')}
-          variant="link"
-          onClick={resetBranch}
-        />
-      )}
-    </p>
-  );
 
-  const branchOptions = repoBranches.map((item, index) => ({
+  const noFeatureBranches = !repoBranches.length;
+  const inputVal = inputs.branch_name;
+  const branchOptions: ComboboxOption[] = repoBranches.map((item, index) => ({
     id: `${index + 1}`,
     label: item,
   }));
+  const selection = branchOptions.find((branch) => branch.label === inputVal);
+  let noOptionsFoundText = null;
+  if (fetchingBranches) {
+    noOptionsFoundText = i18n.t('Loading existing branches…');
+  } else {
+    noOptionsFoundText = (
+      <>
+        {noFeatureBranches
+          ? i18n.t("There aren't any available branches at this time.")
+          : i18n.t('No matching branches found.')}{' '}
+        <Button
+          label={i18n.t('Check again')}
+          variant="link"
+          onClick={doGetBranches}
+        />{' '}
+        {i18n.t('to refresh this list or')}{' '}
+        <Button
+          label={i18n.t('start from a new branch.')}
+          variant="link"
+          onClick={closeBranchForm}
+        />
+      </>
+    );
+  }
 
   return (
     <form onSubmit={handleSubmit} className="slds-form slds-m-bottom--large">
       {isOpen && (
         <>
-          <Trans i18nKey="createProjectHelpText">
-            <p className="slds-p-bottom_medium slds-text-color_weak">
+          <p className="slds-form-element__help slds-m-bottom_small">
+            <Trans i18nKey="createProjectHelpText">
               Projects in Metecho are the high-level features that can be broken
               down into smaller parts when creating Tasks. You can create a new
               project or create a project based on an existing GitHub branch.
               Every project requires a unique project name, which becomes the
-              branch name in Github.
-            </p>
-          </Trans>{' '}
+              branch name in GitHub unless you choose to use an existing branch.
+            </Trans>
+          </p>
           <Checkbox
             id="project-base-branch"
             labels={{
-              label: `${i18n.t('Use existing Github branch')}`,
+              label: `${i18n.t('Use existing GitHub branch')}`,
             }}
             checked={fromBranchChecked}
             className="slds-form-element_stacked slds-p-left_none"
-            onChange={(
-              event: React.ChangeEvent<HTMLInputElement>,
-              { checked }: { checked: boolean },
-            ) => handleChange(checked)}
+            onChange={handleBranchCheckboxChange}
           />
           {fromBranchChecked && (
             <Combobox
               id="combobox-inline-single"
-              isOpen={branchMenuOpen}
```

Comment:
@duggiemitchell I made some changes here mostly to try allowing the ComboBox to handle the open/closed status. Play with it and see what you think?